File: backend/leave/dtos/response/auth_response_dto.py
class AuthResponseDTO:

    @staticmethod
    def login_success(user,access_token,refresh_token):
        return {
            'success': True,
            'access_token': access_token,
            'refresh_token': refresh_token,
            'user': {
                'id': user.id,
                'username': user.username,
                'email': user.email,
                'is_active': user.is_active,
                'is_admin': user.is_staff,
            },
        }
    
    @staticmethod
    def login_error(message='Invalid username or password!'):
        return {
            'success': False,
            'message': message,
        }
    
    @staticmethod
    def logout_success(message='Logout successful!'):
        return {
            'success': True,
            'message': message,
        }

File: frontend/src/dtos/response/authResponseDTO.js
export const parseLoginResponse=(data)=>({

    accessToken: data.accessToken,
    refreshToken: data.refreshToken,
    user:{
        id: data.user.id,
        username : data.user.username,
        email    : data.user.email,
        active   : data.user.is_active,
        isAdmin  : data.user.is_admin,
    }
});